firebase_connector/models/firebase_registration.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2019 Compassion CH (http://www.compassion.ch)
+#    @author: Nicolas Badoux <n.badoux@hotmail.com>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+import logging
+
+try:
+    import firebase_admin
+    from firebase_admin import credentials
+    from firebase_admin import messaging
+except ImportError as e:
+    raise e
+
+from odoo import api, models, fields
+from odoo.tools import config
+
+
+_logger = logging.getLogger(__name__)
+
+try:
+    firebase_credentials = \
+        credentials.Certificate(config.get('google_application_credentials'))
+    firebase_app = firebase_admin.initialize_app(
+        credential=firebase_credentials)
+except (KeyError, ValueError):
+    if not config.get("test_enable"):
+        logging.error("google_application_credentials is not correctly"
+                      " configured in odoo.conf")
+
+
+class FirebaseRegistration(models.Model):
+    """
+    The represent a device that we can send push notification to. It is
+    characterized by the Firebase token and an optional partner.
+    It provides function to send messages to the device.
+    """
+
+    _name = 'firebase.registration'
+    _description = 'Device registered with Firebase Cloud Messaging'
+
+    registration_id = fields.Char(required=True,
+                                  string="Firebase Registration ID")
+    partner_id = fields.Many2one('res.partner', string="Partner")
+    partner_name = fields.Char(related='partner_id.name', readonly=True)
+
+    _sql_constraints = [
+        ('firebase_id_unique',
+         'UNIQUE(registration_id)',
+         'Firebase registration ID should be unique')]
+
+    def name_get(self):
+        res = []
+        for record in self:
+            res.append((record.id, record.registration_id))
+        return res
+
+    def send_message_from_interface(self, context):
+        """
+        Allows to send a notification from Odoo web interface.
+        Button is in the form view but generally hidden.
+        :param context: Context from the view
```

Comment:
Normally the context is automatically passed to the method and accessible via `self.env.context`